### ISSUE TITLE
Strings are too light

### ISSUE THREAD

**Issue by simurai** [primer/github-vscode-theme#19]
Feedback from a DM:

> The contrast of the string is a bit light
>
> ![Screenshot 2020-05-08 at 10 20 47](https://user-images.githubusercontent.com/378023/81387431-a2ec6000-9151-11ea-8ce0-29333c3067cf.png)
>
> compared to the default
>
> ![Screenshot 2020-05-08 at 10 20 08](https://user-images.githubusercontent.com/378023/81387436-a4b62380-9151-11ea-81df-3f978c545b1c.png)

**Comment by minimaxir** [comment]
tbh, inversely the string color is a bit dark in Light mode too. I cannot see any contrast between normal text and strings either.

**Comment by mattaningram** [comment]
This is my first reaction to using this theme as well. Beautiful theme but the light strings makes HTML and CSS harder to read.

**Comment by nscmnto** [comment]
I've been experimenting and it's trickier to make them darker if you want to stick to [Primer](https://primer.style/css/support/color-system) (as other scopes are using darker hues of blue). An alternative would be `#ffab70` which doesn't seem to clash with other colors. However, it introduces a new color to the theme palette, which might be undesired...

**Comment by samuelgozi** [comment]
Agree. That it the first thing I noticed.
Hurts the eyes... I think that if needed we should deviate from Primers colours in this case.
Maybe use a darker shade of the current color?

**Comment by balazs4** [comment]
I am not a designer, but `#fff5b1`  [$yellow-200](https://primer.style/css/support/color-system#yellow) looks good to me:

![image](https://user-images.githubusercontent.com/2198240/81565027-fbfa0500-9398-11ea-83c3-cd8f69eddd4a.png)

/// really nice theme btw :clap:

**Comment by estevanmaito** [comment]
It gets even worse with template literals, as the variables look almost the same as the surrounding string.

GitHub Dark
![github-dark](https://user-images.githubusercontent.com/8527573/81584459-28a62080-9389-11ea-9818-2d0b455eed0e.png)

Dracula
![dracula-dark](https://user-images.githubusercontent.com/8527573/81584627-5d19dc80-9389-11ea-84e4-d6a81d39e671.png)

**Comment by simurai** [comment]
> tbh, inversely the string color is a bit dark in Light mode too.

Yeah, it's a tricky question. How much should this "GitHub theme" try to mimic the syntax highlighting on github.com? Below an example and a screenshot from VS Code. It doesn't have to be 100% the same, but would still be nice to stay somewhat close. Not that github.com's syntax highlighting is perfect, but people should feel familiar. :grimacing:

https://github.com/primer/github-vscode-theme/blob/71691fbdde883083ad95b60d454b689e35a43f47/src/index.js#L10-L13

<img width="272" alt="CleanShot 2020-05-12 at 18 53 00@2x" src="https://user-images.githubusercontent.com/378023/81669918-dc8ad700-9481-11ea-92b7-8fcc1b250fae.png">

As for the dark version.. we might can be less strict since that doesn't exist for github.com.

> I've been experimenting and it's trickier to make them darker if you want to stick to [Primer](https://primer.style/css/support/color-system) (as other scopes are using darker hues of blue).

Yeah, `blue[7]` is still pretty light and `blue[6]` already gets used a lot elsewhere. Here a mix in between:

<img width="282" alt="CleanShot 2020-05-12 at 18 51 50@2x" src="https://user-images.githubusercontent.com/378023/81669806-b107ec80-9481-11ea-8640-688ec3b8e75d.png">

Maybe as quick fix before considering a bigger change.

**Comment by loreina** [comment]
I feel a lighter blue is too close to $blue-300 `#79b8ff` and can be hard to read in the same line:

![image](https://user-images.githubusercontent.com/23108291/81753942-ecef8180-9482-11ea-80f8-d6b535421686.png)

I'm using $yellow-200 `#fff5b1` as @balazs4 suggested as a temporary fix:

![image](https://user-images.githubusercontent.com/23108291/81753957-f973da00-9482-11ea-8205-78e8bdeaee75.png)

**Comment by paramaggarwal** [comment]
As a user, I come to use this theme because I want to feel at home when switching between GitHub web and VSCode. I think we should honour this expectation and try to stay close to the GitHub web theme.

**Comment by simurai** [comment]
> I'm using $yellow-200 `#fff5b1` as @balazs4 suggested as a temporary fix:

👍 If anyone else wants to test it out, here the config you can add to your `settings.json`:

```
  "editor.tokenColorCustomizations": {
    "[GitHub Dark]": {
      "strings": "#fff5b1"
    }
  },
```

![image](https://user-images.githubusercontent.com/378023/81790759-ce4fc000-9540-11ea-941a-46c8d94154ae.png)

**Comment by minimaxir** [comment]
The corresponding TextMate rule will have to be added too:

```
"[GitHub Dark]": {
      "strings": "#fff5b1",
      "textMateRules": [
        {
          "scope": "punctuation.definition.string",
          "settings": {
            "foreground": "#fff5b1"
          }
        }
      ]
    }
```

I've been testing that out and it works fine.

**Comment by MrSunshyne** [comment]
> As a user, I come to use this theme because I want to feel at home when switching between GitHub web and VSCode. I think we should honour this expectation and try to stay close to the GitHub web theme.

Not at the expense of readability

**Comment by paramaggarwal** [comment]
@MrSunshyne Your claim is that readability is lacking on GitHub website? Because my point was about sticking to the GitHub website design as much as possible.

**Comment by estevanmaito** [comment]
> 
> 
> @MrSunshyne Your claim is that readability is lacking on GitHub website? Because my point was about sticking to the GitHub website design as much as possible.

I think you're missing the point of this issue. GitHub uses a light theme, and as pointed earlier (https://github.com/primer/github-vscode-theme/issues/19#issuecomment-627240537) there is no intent on changing it.

We're trying to find a way to improve readability for the **Dark Theme**, and with this particular theme, there is more room for changes, as GitHub doesn't have a dark mode, so there is no standard for it.

**Comment by balazs4** [comment]
> We're trying to find a way to improve readability for the **Dark Theme**, and with this particular theme, there is more room for changes, as GitHub doesn't have a dark mode, so there is no standard for it.

And how about the Github Mobile (iOS) dark mode?

![image](https://user-images.githubusercontent.com/2198240/82258995-27fa2500-995b-11ea-8774-35e4b851d24a.jpeg)

Same color palette (?), but they are slightly different assigned to the tokens.

I personally find the mobile version somehow cleaner. 🙄

What do you think?

**Comment by estevanmaito** [comment]
You're right, mobile apps (Android uses the same theme as iOS) already have a dark theme, and the color palette seems the same, but better applied.

**Comment by paramaggarwal** [comment]
Oh good idea - we can use the mobile app dark mode as our reference 👍 - again my only point is that because this is an official GitHub theme, it needs to follow GitHub conventions on other platforms.

**Comment by simurai** [comment]
> And how about the Github Mobile (iOS) dark mode?

Yeah, that's a great idea. Let's see if we can :flashlight: :mag: spy on their dark theme and steal a few hex values.

**Comment by this-fifo** [comment]
There's also a [dark theme](https://github.com/desktop/desktop/blob/development/app/styles/themes/_dark.scss) available for the GitHub Desktop app

**Comment by CatsMiaow** [comment]
- VS Code v1.45.1
- GitHub Theme v1.1.2 dark

Is there any reason why the color of the const variable is similar to the color of the text?
![image](https://user-images.githubusercontent.com/1300172/82532338-1e9bd300-9b7c-11ea-8d58-1cdfa3bc59bf.png)

**Comment by MrSunshyne** [comment]
> @MrSunshyne Your claim is that readability is lacking on GitHub website? Because my point was about sticking to the GitHub website design as much as possible.

@paramaggarwal  Yep that is my claim. I would 100% prefer loading a project in vscode/webstorm darkmode before doing any serious code review. I didn't even know I was doing that until I thought about it after reading your question. I can glance at the code on github.com, but not for long. Also, to be clear, this is my opinion and I understand thousands of people do code review on github since years and it's working perfectly fine for them. 

@estevanmaito since there is no standard for it, it's probably makes inserting new variations(removing while text) more acceptable?

**Comment by estevanmaito** [comment]
@MrSunshyne actually we found that there is already a Dark Mode for GitHub, mainly used on mobile, and it actually looks good https://github.com/primer/github-vscode-theme/issues/19#issuecomment-630435353

It would make sense using it.

**Comment by icebeam030** [comment]
This might be irrelevant to this issue but I feel using #fff as white is too bright in a dark theme and kind of hurts my eyes.
